src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -81,13 +97,239 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
     dTotalStress_dPressure[4] = 0;
     dTotalStress_dPressure[5] = 0;
 
+    real64 dPorosity_dPressure;
+    real64 dPorosity_dVolStrain;
     m_porosityUpdate.updateFromPressureAndStrain( k,
                                                   q,
-                                                  deltaPressure,
+                                                  deltaFluidPressure,
                                                   strainIncrement,
                                                   dPorosity_dPressure,
                                                   dPorosity_dVolStrain );
 
+    real64 const porosity = m_porosityUpdate.getPorosity( k, q );
+    real64 const porosityOld = m_porosityUpdate.getOldPorosity( k, q );
+    real64 const porosityInit = m_porosityUpdate.getInitialPorosity( k, q );
+
+    // Compute body force vector and its derivatives w.r.t. to
+    // volumetric strain and pressure. The following assumption
+    // are made at the moment:
+    // 1. dMixtureDens_dVolStrainIncrement is neglected,
+    // 2. grains are assumed incompressible
+    real64 const mixtureDensity = ( 1.0 - porosity ) * solidDensity + porosity * fluidDensity;
+    real64 const initialMixtureDensity = ( 1.0 - porosityInit ) * solidDensity + porosityInit * initialFluidDensity;
+    real64 const mixtureDensityIncrement = mixtureDensity - initialMixtureDensity;
+
+    real64 const dMixtureDens_dVolStrainIncrement = 0.0;
+    real64 const dMixtureDens_dPressure = dPorosity_dPressure * ( -solidDensity + fluidDensity )
+                                          + porosity * dFluidDensity_dPressure;
+
+    if( gravityAcceleration > 0.0 )
```

Comment:
could you start this if statement at line 117?

Reply:
Done